indexing-service/src/main/java/org/apache/druid/indexing/common/TaskToolbox.java

```diff
@@ -287,6 +295,11 @@ public QueryRunnerFactoryConglomerate getQueryRunnerFactoryConglomerate()
     return queryRunnerFactoryConglomerateProvider.get();
   }
 
+  public DruidProcessingConfig getProcessingConfig()
```

Comment:
Based on 
```
   * Using Provider, not {@link DruidProcessingConfig} directly, because it may not be available for task
   * types that do not run queries.
   */
```
Does it make sense to mark this method nullable?

Reply:
I'm not sure, I don't remember if a `Provider` with nothing to provide returns `null` or throws an exception. Fwiw, the reason this isn't annotated is because I was copying the design of `getQueryRunnerFactoryConglomerate()`, which also isn't annotated. I figured it was a good one to copy because like `DruidProcessingConfig` it is also query-related.